From: David Milburn <dmilburn@redhat.com>
Date: Thu, 23 Jul 2009 13:20:15 -0500
Subject: [ata] ahci: add IDs for Ibex Peak ahci controllers
Message-id: 20090723182015.GA6616@dhcp-210.hsv.redhat.com
O-Subject: [RHEL5.4 PATCH] ahci: add device IDs for Ibex Peak sata ahci controllers.
Bugzilla: 513067
RH-Acked-by: Jeff Garzik <jgarzik@redhat.com>
RH-Acked-by: Peter Martuccelli <peterm@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Add Device IDs for Ibex Peak SATA AHCI controllers.

A -159.el5 test kernel built with this patch has
been verified by Intel, this resolves BZ 513067.

Upstream patch has been submitted to linux-ide and
is currently under review. Please review and ACK.

Thanks,
David

 drivers/ata/ahci.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/drivers/ata/ahci.c b/drivers/ata/ahci.c
index 706ad81..069eb5e 100644
--- a/drivers/ata/ahci.c
+++ b/drivers/ata/ahci.c
@@ -479,10 +479,14 @@ static const struct pci_device_id ahci_pci_tbl[] = {
 	{ PCI_VDEVICE(INTEL, 0x502b), board_ahci }, /* Tolapai */
 	{ PCI_VDEVICE(INTEL, 0x3a05), board_ahci }, /* ICH10 */
 	{ PCI_VDEVICE(INTEL, 0x3a25), board_ahci }, /* ICH10 */
+	{ PCI_VDEVICE(INTEL, 0x3b22), board_ahci }, /* PCH AHCI */
+	{ PCI_VDEVICE(INTEL, 0x3b23), board_ahci }, /* PCH AHCI */
 	{ PCI_VDEVICE(INTEL, 0x3b24), board_ahci }, /* PCH RAID */
 	{ PCI_VDEVICE(INTEL, 0x3b25), board_ahci }, /* PCH RAID */
+	{ PCI_VDEVICE(INTEL, 0x3b29), board_ahci }, /* PCH AHCI */
 	{ PCI_VDEVICE(INTEL, 0x3b2b), board_ahci }, /* PCH RAID */
 	{ PCI_VDEVICE(INTEL, 0x3b2c), board_ahci }, /* PCH RAID */
+	{ PCI_VDEVICE(INTEL, 0x3b2f), board_ahci }, /* PCH AHCI */
 
 	/* JMicron 360/1/3/5/6, match class to avoid IDE function */
 	{ PCI_VENDOR_ID_JMICRON, PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID,
